### DISCUSSION TITLE
Manual Data Entry

### DISCUSSION THREAD

**Discussion by asadleo94** [cloud-carbon-footprint/cloud-carbon-footprint#1165] [Q&A] [answered]
Hi, my organization doesn't allow me to connect to the app using the API. I want to estimate CO2 emissions. Can someone please help me and let me know how to manually add data? Your answer would be greatly appreciated.

**Comment by 4upz** [comment, marked as answer]
Hi @asadleo94 have you tried out the Lookup Table approach? It allows you to specify a list of services as a manual input, and will provide mapped estimated values for energy usage and emissions for 1 unit of usage for that service. You can then multiply the resulting values by your actual usage to get results.

You can learn more about it here: https://www.cloudcarbonfootprint.org/docs/creating-a-lookup-table

**Comment by asadleo94** [comment]
Hi @4upz , Thanks for your response and help. I was really missing this feature. It would mean a lot. Do you have any idea how I can visualize it on the dashboard?  could you please suggest me a workaround?

**Reply by 4upz** [reply]
Hi @asadleo94 I'm glad that the LookUp table method was able to help. As for visualizing the results, we unfortunately don't have a way to input manual data into the dashboard. If you are comfortable with JS/TS and don't mind tinkering a little, you could try the following workarounds:

- Convert the csv files with the calculated results to JSON, and modify the `packages/client` to read from the JSON file. The data would need to be formatted in the same API response structure. (See EstimationResults in `packages/common/src/EstimationResult.ts`)
- Similar to the first option, you can convert the csv results to JSON, and modify the API to return this data instead. It would still need to match the proper response structure

If you aren't comfortable with those approaches, I'd recommend the easier approach of loading the csv results into a BI tool such as [Google Looker Studio](https://lookerstudio.google.com/) or [Microsoft Power BI](https://powerbi.microsoft.com/en-us/) to import the csv file as a data set so that you can create visualizations from it.

**Comment by asadleo94** [comment]
Hi  @4upz and @mfulleratlassian,

I need your support. I am attempting to calculate the carbon footprint of AWS S3 based on usage through the lookup function. I have tried it, and it is providing results in terms of CO2e and KWH. However, I am having trouble mapping these results to my storage usage. For example, some S3 buckets have 100GB of data, and others have more or less. While I understand that EC2 instances provide hourly KWH, which I can multiply with actual usage, I am a bit confused about the output result from the lookup function of S3, Lambda, or other services and what it actually represents.

Thanks for your support and help.
